build.ps1

```diff
@@ -242,13 +242,18 @@ Verify-DotnetExists
 $configFile = "$repoPath\NuGet.config"
 if ($ARM64)
 {
-    $configFile = "$repoPath\NuGet.internal.config"   
+    $configFile = "$repoPath\NuGet.internal.config"
 }
 
 if (!$SkipBuild)
 {
     if (!$SkipCleanAndRestore)
     {
+        # The $env:NUGET_PACKAGES variable overrides the globalPackagesFolder set in nuget.config.
+        # This clears it out temporarily for proper nuget restore behavior.
```

Comment:
Why? The point of NUGET_PACKAGES is that a user can direct where the packages are restored. What's the specific error that happens without this change?

Reply:
The problem is that we point $(PackagesDir) to the $(EnlistmentRoot)\packages so builds will fail to resolve when reading from that property. The alternative fix wouild be to respect using NUGET_PACKAGES everywhere.

Reply:
The error we get when building ARM64 is:

LINK : fatal error LNK1104: cannot open file 'E:\ClrInstrumentationEngineGitHub\packages\microsoft.visualstud
       io.arm64.sdk\17.0.0\lib\mscoree.lib' [E:\ClrInstrumentationEngineGitHub\src\InstrumentationEngine\Instrumenta
       tionEngine.vcxproj]